### DISCUSSION TITLE
Can we change some `constexpr` functions to `consteval` in C++20 and later?

### DISCUSSION THREAD

**Discussion by frederick-vs-ja** [microsoft/STL#3604] [Q&A] [answered]
Some `constexpr` functions in the standard library always return the same constant value. It seems that we can change some of them to `consteval` in C++20 and later modes.

It seems that the only difference happens on taking address. However, most standard library function are made non-addressable by WG21-P0551R3 in C++20, so ill-formedness on taking address seems OK.

**Comment by StephanTLavavej** [comment, marked as answer]
@CaseyCarter says "this seems very scary, purely from a new-compiler-technology point of view". As Casey is historically the most aggressive "yay new compiler features let's use them and they'll be awesome! :unicorn:" maintainer, and STL is the most cautious and curmudgeonly "ugh new compiler features, that looks like a great way to discover exotic compiler bugs :skull_and_crossbones:", caution seems especially warranted here :joy_cat:
